catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java

```diff
@@ -388,153 +459,76 @@ public void destroy(int code) {
-            currentContentTypeMapping = currentContentTypeMapping.trim();
-            if (!currentContentTypeMapping.equals(previousContentTypeMapping)) {
-                LOGGER.debug("{}: The content type has been updated from {} to {}.",
-                        cswSourceConfiguration.getId(), previousContentTypeMapping,
-                        currentContentTypeMapping);
-                contentTypes.clear();
+        // Filter Configuration Map
+        Map<String, Object> configs = filter(configuration);
+
+        // Run Filtered Consumers
+        for (Map.Entry<String, Object> entry : configs.entrySet()) {
+            String s = entry.getKey();
+            Consumer c = consumerMap.get(s);
+            if (c != null) {
+                c.accept(configs.get(s));
             }
-        } else {
-            currentContentTypeMapping = CswRecordMetacardType.CSW_TYPE;
         }
 
-        cswSourceConfiguration.setContentTypeMapping(currentContentTypeMapping);
+        configureCswSource();
 
-        LOGGER.debug("{}: Current content type mapping: {}.", cswSourceConfiguration.getId(),
-                currentContentTypeMapping);
+        initClientFactory();
+    }
 
-        Integer newPollInterval = (Integer) configuration.get(POLL_INTERVAL_PROPERTY);
+    private Map<String, Object> filter(Map<String, Object> configuration) {
+        // Filter blank strings from configuration map
+        Map<String, Object> configs = configuration.entrySet()
+                .stream()
+                .filter(p -> p.getValue() instanceof String
+                        && StringUtils.isNotBlank((String) p.getValue()))
+                .collect(Collectors.toMap(p -> p.getKey(), p -> p.getValue()));
 
-        if (newPollInterval != null && !newPollInterval
-                .equals(cswSourceConfiguration.getPollIntervalMinutes())) {
-            LOGGER.debug("Poll Interval was changed for source {}.",
-                    cswSourceConfiguration.getId());
-            cswSourceConfiguration.setPollIntervalMinutes(newPollInterval);
-            availabilityPollFuture.cancel(true);
-            setupAvailabilityPoll();
-        }
+        // Filter false booleans
+        configs.putAll(configuration.entrySet()
+                .stream()
+                .filter(p -> (p.getValue() instanceof Boolean && !((Boolean) p.getValue())))
+                .collect(Collectors.toMap(p -> p.getKey(), p -> p.getValue())));
 
-        String cswUrlProp = (String) configuration.get(CSWURL_PROPERTY);
-        if (StringUtils.isNotBlank(cswUrlProp) &&
-                !cswUrlProp.equals(cswSourceConfiguration.getCswUrl())) {
-            cswSourceConfiguration.setCswUrl(cswUrlProp);
-        }
-        configureCswSource();
+        // Filter null integers
+        configs.putAll(configuration.entrySet()
+                .stream()
+                .filter(p -> (p.getValue() instanceof Integer && (p.getValue()) != null))
```

Comment:
You could just use `p.getValue() instanceof Integer` because that will return false it `p.getValue()` is null.